stan/math/prim/scal/fun/common_type.hpp

```diff
@@ -0,0 +1,24 @@
+#ifndef STAN_MATH_PRIM_SCAL_FUN_COMMON_TYPE_HPP
+#define STAN_MATH_PRIM_SCAL_FUN_COMMON_TYPE_HPP
+
+#include <boost/math/tools/promotion.hpp>
+
+namespace stan {
+  namespace math {
+    /**
+     * Struct which calculates type promotion given two types.
+     *
+     * <p>This is the base implementation for scalar types.
+     *
+     * @tparam T1 scalar type, either int, double, or stan::math::var
```

Comment:
Why the restriction on types?  Will this not work when we start using `fvar<T>` for forward mode?  When I said all types would be int, double or autodiff variables, I meant all the different autodiff variables, but that if an autodiff variable was used, they'd all be the same autodiff variables.  That is, you wouldn't get an `fvar<var>` and a `var`.